tests/test_sklearn_nearest_neighbour_converter.py

```diff
@@ -73,88 +94,98 @@ def test_model_knn_regressor2_2(self):
         self.assertIsNotNone(model_onnx)
         dump_data_and_model(
             X.astype(numpy.float32)[:7],
-            model,
-            model_onnx,
-            basename="SklearnKNeighborsRegressorMetricCityblock-OneOffArray",
-            allow_failure="StrictVersion(onnxruntime.__version__) <= "
-            "StrictVersion('0.2.1') or "
-            "StrictVersion(onnx.__version__) == StrictVersion('1.4.1')",
-        )
+            model, model_onnx,
+            basename="SklearnKNeighborsRegressorMetricCityblock")
 
     @unittest.skipIf(not onnx_built_with_ml(),
                      reason="Requires ONNX-ML extension.")
+    @unittest.skipIf(
+        StrictVersion(onnxruntime.__version__) < StrictVersion("0.5.0"),
+        reason="not available")
     def test_model_knn_classifier_binary_class(self):
         model, X = self._fit_model_binary_classification(
             KNeighborsClassifier())
         model_onnx = convert_sklearn(
             model,
             "KNN classifier binary",
-            [("input", FloatTensorType([None, 3]))],
+            [("input", FloatTensorType([None, X.shape[1]]))],
         )
         self.assertIsNotNone(model_onnx)
         dump_data_and_model(
             X.astype(numpy.float32),
-            model,
-            model_onnx,
-            basename="SklearnKNeighborsClassifierBinary-OneOffArray",
-            allow_failure="StrictVersion(onnx.__version__) "
-            "== StrictVersion('1.1.2') or "
-            "StrictVersion(onnxruntime.__version__) <= StrictVersion('0.2.1') "
-            "or StrictVersion(onnx.__version__) == StrictVersion('1.4.1')",
-        )
+            model, model_onnx,
+            basename="SklearnKNeighborsClassifierBinary")
 
+    @unittest.skipIf(True, reason="later")
     @unittest.skipIf(not onnx_built_with_ml(),
                      reason="Requires ONNX-ML extension.")
+    @unittest.skipIf(
+        StrictVersion(onnxruntime.__version__) < StrictVersion("0.5.0"),
+        reason="not available")
     def test_model_knn_classifier_multi_class(self):
         model, X = self._fit_model_multiclass_classification(
             KNeighborsClassifier())
         model_onnx = convert_sklearn(
             model,
             "KNN classifier multi-class",
-            [("input", FloatTensorType([None, 3]))],
+            [("input", FloatTensorType([None, X.shape[1]]))],
         )
         self.assertIsNotNone(model_onnx)
         dump_data_and_model(
             X.astype(numpy.float32),
+            model, model_onnx,
+            basename="SklearnKNeighborsClassifierMulti")
+
+    @unittest.skipIf(not onnx_built_with_ml(),
+                     reason="Requires ONNX-ML extension.")
+    @unittest.skipIf(
+        StrictVersion(onnxruntime.__version__) < StrictVersion("0.5.0"),
+        reason="not available")
+    def test_model_knn_classifier_multi_class_string(self):
+        model, X = self._fit_model_multiclass_classification(
```

Comment:
We may want to clean up this unit test and use utility functions defined in test_utils.py later.